tools/codegen/gen.py

```diff
@@ -241,14 +227,6 @@ def func(f: NativeFunction) -> Optional[str]:
                 # Only tensor like arguments are eligible
                 device_of = next((f'{a.name}' for a in candidate_args if a.type.is_tensor_like()), None)
 
-                # See Note [Byte-for-byte compatibility]
-                # I wasn't able to figure out the internal logic for
-                # these device guards
-                if str(f.func.name) == "_thnn_fused_lstm_cell_backward":
-                    device_of = "cx"
```

Comment:
It seems like in the previous codegen, these two functions arbitrarily used these two variables to create a device guard instead of using the first "tensor_like" argument in the function.